spec/models/reply_spec.rb

```diff
@@ -187,4 +187,22 @@
       include_examples 'has_edit_audits', method
     end
   end
+
+  describe "#reply_order" do
+    let(:post) { create(:post) }
+
+    it "puts replies in the correct order" do
+      first_reply = create(:reply, post: post)
+      second_reply = create(:reply, post: post)
+      third_reply = create(:reply, post: post)
+      expect(post.replies.ordered).to eq([first_reply, second_reply, third_reply])
+    end
+
+    it "puts replies in the correct order even if they are created in the wrong order" do
+      first_reply = create(:reply, post: post)
+      second_reply = Timecop.freeze(first_reply.created_at - 5.seconds) { create(:reply, post: post) }
+      third_reply = Timecop.freeze(first_reply.created_at - 3.seconds) { create(:reply, post: post) }
+      expect(post.replies.ordered).to eq([first_reply, second_reply, third_reply])
+    end
+  end
```

Comment:
Could we have another test for:

```ruby
it "orders replies by reply order not ID" do
  reply1 = create(:reply, post: post)
  reply2 = create(:reply, post: post)
  reply3 = create(:reply, post: post)
  reply2.update_columns(reply_order: 2)
  reply3.update_columns(reply_order: 1)
  expect(post.replies.ordered).to eq([reply1, reply3, reply2])
end
```

The second test would work on master (well, if we added a scope for `ordered`), since that orders by ID – it doesn't actually test that we're now ordering by _reply order_ instead of _ID_.

Reply:
this test was originally supposed to do that and then I discovered that it wouldn't let me pass reply_order to the factory. that sounds like a way to do it!